azurelinuxagent/common/utils/archive.py

```diff
@@ -41,71 +41,89 @@
 MAX_ARCHIVED_STATES = 50
 
 CACHE_PATTERNS = [
-    re.compile("^(.*)\.(\d+)\.(agentsManifest)$", re.IGNORECASE), # pylint: disable=W1401
-    re.compile("^(.*)\.(\d+)\.(manifest\.xml)$", re.IGNORECASE), # pylint: disable=W1401
-    re.compile("^(.*)\.(\d+)\.(xml)$", re.IGNORECASE) # pylint: disable=W1401
+    re.compile(r"^(.*)\.(\d+)\.(agentsManifest)$", re.IGNORECASE),
+    re.compile(r"^(.*)\.(\d+)\.(manifest\.xml)$", re.IGNORECASE),
+    re.compile(r"^(.*)\.(\d+)\.(xml)$", re.IGNORECASE)
 ]
 
+GOAL_STATE_PATTERN = re.compile(r"^(.*)GoalState\.(\d+)\.xml$", re.IGNORECASE)
```

Comment:
we do not want the "(.*)" before "GoalState", do we?

Reply:
This regex is compared to the file's full path, e.g. `/var/lib/waagent/GoalState.N.xml` (or a random tmp directory while testing).

Reply:
if that is the case then try this instead:

`r"^(.*)/GoalState\.(\d+)\.xml$"`

Reply:
Ok, will do that.